recipes/pulseaudio/meson/conanfile.py

```diff
@@ -0,0 +1,147 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import cross_building
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
+from conan.tools.files import copy, get, rm, rmdir
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.meson import Meson, MesonToolchain
+from conan.tools.layout import basic_layout
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class PulseAudioConan(ConanFile):
+    name = "pulseaudio"
+    description = "PulseAudio is a sound system for POSIX OSes, meaning that it is a proxy for sound applications."
+    topics = ("sound",)
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://pulseaudio.org/"
+    license = "LGPL-2.1"
+
+    package_type = "shared-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_glib": [True, False],
+        "with_fftw": [True, False],
+        "with_x11": [True, False],
+        "with_openssl": [True, False],
+        "with_dbus": [True, False],
+    }
+    default_options = {
+        "with_glib": False,
+        "with_fftw": False,
+        "with_x11": True,
+        "with_openssl": True,
+        "with_dbus": False,
+    }
+
+    def config_options(self):
+        if self.settings.os not in ['Linux', 'FreeBSD']:
+            del self.options.with_x11
+
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+        if not self.options.with_dbus:
+            del self.options.with_fftw
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("libgettext/0.22")
+        self.requires("libiconv/1.17")
+        self.requires("libsndfile/1.2.2")
+        if self.options.with_glib:
+            self.requires("glib/2.78.1")
+        if self.options.get_safe("with_fftw"):
+            self.requires("fftw/3.3.10")
+        if self.options.get_safe("with_x11"):
+            self.requires("xorg/system")
+        if self.options.with_openssl:
+            self.requires("openssl/[>=1.1 <4]")
+        if self.options.with_dbus:
+            self.requires("dbus/1.15.8")
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("pulseaudio supports only linux currently")
```

Comment:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} recipe is only compatible with Linux right now. Contributions are welcome.")
```

The changelog says is possible to build on Windows now.

Reply:
I got this error `..\src\meson.build:63:0: ERROR: Assert failed: Prefix is not absolute: "D:/a/_temp/.c2/p/b/pulse0b1765b681bfa/p"` https://github.com/ericLemanissier/proof-of-conan/actions/runs/7585767265/job/20666352739#step:7:306

As the package does not support windows currently https://github.com/conan-io/conan-center-index/blob/master/recipes/pulseaudio/all/conanfile.py#L78-L79 I chose not to loose time on this

Reply:
No problem at all, I'm sure trying to support Windows will be a challenge. It can be done later for sure.